### ISSUE TITLE
Refactoring aware review

### ISSUE THREAD

**Issue by ttozser** [Reviewable/Reviewable#719]
It is painful to review a PR that contains the results of IDE refactors. For example, moving a method into another file or renaming a method. 
Among a large number of automatic changes, it is hard to spot if something else was changed manually.

It would be useful if reviewable could say that
* this method was moved without modification to this other place
* this method was renamed and this caused changes in these twenty-three files. Then you can approve all of them in a single click.

I would primarily use this feature with java.

You can find a similar solution here
https://github.com/rodrigo-brito/refactoring-aware-review
https://github.com/aserg-ufmg/RefDiff

**Comment by pkaminski** [comment]
Yep, I agree that this would be extremely useful.  I generally shy away from doing language-specific stuff, though, unless the benefit of doing so is really huge.  I suspect that most of the refactorings could be detected almost as well just by looking for block moves at a character level, as requested all the way back in #5.

Still, it's worth keeping this feature request open in case it proves to be a better approach.

**Comment by ttozser** [comment]
Yes, detecting if a method was moved is much easier and would be very useful on its own.

I am not really sure that detecting method renames with a usable error rate is possible without parsing the code.
However, I am aware that this is a very complex request that may not worth adding to the product.
